modules/ppcp-save-payment-methods/resources/js/add-payment-method.js

```diff
@@ -0,0 +1,75 @@
+import {
+    getCurrentPaymentMethod,
+    ORDER_BUTTON_SELECTOR,
+    PaymentMethods
+} from "../../../ppcp-button/resources/js/modules/Helper/CheckoutMethodState";
+
+import {setVisible} from "../../../ppcp-button/resources/js/modules/Helper/Hiding";
+import {loadPaypalJsScript} from "../../../ppcp-button/resources/js/modules/Helper/ScriptLoading";
+
+loadPaypalJsScript(
+    {
+        clientId: ppcp_add_payment_method.client_id,
+        merchantId: ppcp_add_payment_method.merchant_id,
+        dataUserIdToken: ppcp_add_payment_method.id_token,
+    },
+    {
+        createVaultSetupToken: async () => {
+            const response = await fetch(ppcp_add_payment_method.ajax.create_setup_token.endpoint, {
+                method: "POST",
+                credentials: 'same-origin',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                    nonce: ppcp_add_payment_method.ajax.create_setup_token.nonce,
+                })
+            })
+
+            const result = await response.json()
+
+            if(result.data.id) {
+                return result.data.id
+            }
+        },
+        onApprove: async ({ vaultSetupToken }) => {
+            console.log(vaultSetupToken)
+
+            const response = await fetch(ppcp_add_payment_method.ajax.create_payment_token.endpoint, {
+                method: "POST",
+                credentials: 'same-origin',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                    nonce: ppcp_add_payment_method.ajax.create_payment_token.nonce,
+                    vault_setup_token: vaultSetupToken,
+                })
+            })
+
+            const result = await response.json()
+            console.log(result)
+        },
+        onError: (error) => {
+            console.error(error)
+        }
+    },
+    `#ppc-button-${PaymentMethods.PAYPAL}-save-payment-method`
+);
+
+
+
+const init = () => {
+    setVisible(ORDER_BUTTON_SELECTOR, getCurrentPaymentMethod() !== PaymentMethods.PAYPAL);
+    setVisible(`#ppc-button-${PaymentMethods.PAYPAL}-save-payment-method`, getCurrentPaymentMethod() === PaymentMethods.PAYPAL);
+}
+
+document.addEventListener(
+    'DOMContentLoaded',
+    () => {
+        jQuery(document.body).on('click init_add_payment_method', '.payment_methods input.input-radio', function () {
```

Comment:
Maybe it would be simpler to use `updated_checkout payment_method_selected` like in `CheckoutBootstrap`?

Reply:
I don't foresee the side effects of this change. I'll leave to @Dinamiko.

Reply:
@AlexP11223 I tried changing from `click init_add_payment_method` to `updated_checkout payment_method_selected` but it does not work. 

I added the same as WC is using for switching saved payment methods: https://github.com/woocommerce/woocommerce/blob/2f7bd83ae229a999456e3e713af4d4a98e2bd28d/plugins/woocommerce/client/legacy/js/frontend/add-payment-method.js#L12

Reply:
it should work like this, without the 2nd parameter: `jQuery(document.body).on('updated_checkout payment_method_selected', () => { .......`

But I guess it should be fine too if WC is using it.